src/server/authserver/Server/AuthSocket.cpp

```diff
@@ -364,188 +364,180 @@ bool AuthSocket::_HandleLogonChallenge()
-                            pkt << uint8(WOW_FAIL_UNLOCKABLE_LOCK);
-                            locked = true;
-                        }
-                        else
-                            LOG_DEBUG("network", "[AuthChallenge] Account country matches");
-                    }
-                    else
-                        LOG_DEBUG("network", "[AuthChallenge] IP2NATION Table empty");
+                    LOG_DEBUG("network", "[AuthChallenge] Account country differs.");
+                    pkt << uint8(WOW_FAIL_UNLOCKABLE_LOCK);
+                    SendAuthPacket();
+                    return true;
                 }
             }
+        }
+    }
 
-            uint8 securityFlags = 0;
-            _totpSecret = fields[7].GetBinary();
+    // If the account is banned, reject the logon attempt
+    if (_accountInfo.IsBanned)
+    {
+        if (_accountInfo.IsPermanenetlyBanned)
+        {
+            pkt << uint8(WOW_FAIL_BANNED);
+            LOG_DEBUG("server.authserver.banned", "'%s:%d' [AuthChallenge] Banned account %s tried to login!", ipAddress.c_str(), port, _accountInfo.Login.c_str());
+        }
+        else
+        {
+            pkt << uint8(WOW_FAIL_SUSPENDED);
+            LOG_DEBUG("server.authserver.banned", "'%s:%d' [AuthChallenge] Temporarily banned account %s tried to login!", ipAddress.c_str(), port, _accountInfo.Login.c_str());
+        }
 
-            // Check if a TOTP token is needed
-            if (!_totpSecret || !_totpSecret.value().empty())
-            {
-                securityFlags = 4;
+        SendAuthPacket();
+        return true;
+    }
 
-                if (auto const& secret = sSecretMgr->GetSecret(SECRET_TOTP_MASTER_KEY))
-                {
-                    bool success = acore::Crypto::AEDecrypt<acore::Crypto::AES>(*_totpSecret, *secret);
-                    if (!success)
-                    {
-                        pkt << uint8(WOW_FAIL_DB_BUSY);
-                        LOG_ERROR("server.authserver", "[AuthChallenge] Account '%s' has invalid ciphertext for TOTP token key stored", _login.c_str());
-                        locked = true;
-                    }
-                }
-            }
+    uint8 securityFlags = 0;
 
-            if (!locked)
-            {
-                //set expired bans to inactive
-                LoginDatabase.DirectExecute(LoginDatabase.GetPreparedStatement(LOGIN_UPD_EXPIRED_ACCOUNT_BANS));
+    // Check if a TOTP token is needed
+    if (!fields[7].IsNull())
+    {
+        LOG_DEBUG("server.authserver", "[AuthChallenge] Account '%s' using TOTP", _accountInfo.Login.c_str());
 
-                // If the account is banned, reject the logon attempt
-                stmt = LoginDatabase.GetPreparedStatement(LOGIN_SEL_ACCOUNT_BANNED);
-                stmt->setUInt32(0, fields[0].GetUInt32());
+        securityFlags = 4;
+        _totpSecret = fields[7].GetBinary();
```

Comment:
```suggestion
        _totpSecret = fields[9].GetBinary();
```

Reply:
@Winfidonarleyan did a bit of testing, changing to fields[9] instead of fields[7] fixes the issue with the TOTP.